slither/detectors/statements/incorrect_strict_equality.py

```diff
@@ -72,6 +72,14 @@ class IncorrectStrictEquality(AbstractDetector):
     def is_direct_comparison(ir: Operation) -> bool:
         return isinstance(ir, Binary) and ir.type == BinaryType.EQUAL
 
+    @staticmethod
+    def is_not_comparing_addresses(ir: Binary) -> bool:
+        """
+        Comparing addresses strictly should not be flagged.
+        """
+        addr = ElementaryType("address")
+        return ir.variable_left.type != addr or ir.variable_right.type != addr
```

Comment:
If something like `Contract(address(0x0)) == Contract(address(0x0))` is used, `is_not_comparing_addresses` won't filter it out. We would need to add something like `isinstance(ir.variable_left.type, UserDefinedType) and isinstance(ir.variable_left.type.type, Contract)`. It'd be helpful to add a test case for both the address and contract case so that we can be sure of the modifications and prevent regressions in the future.

Reply:
alright, will do that